src/hiccup/util.clj

```diff
@@ -52,6 +52,26 @@
   String
   (to-uri [s] (URI. s)))
 
+(deftype RawString [^String s]
+  Object
+  (^String toString [this] s)
+  (^boolean equals [this other]
+    (cond
+      (instance? String other) (= s other)
```

Comment:
We probably don't want Strings to be equal to RawStrings. I'd rather they were more opaque to give us more options in future as to what the intermediate form looks like.